script.js

```diff
@@ -1,4 +1,206 @@
+  var numAcesInHand = 0;
+  var sum = 0;
+  var i = 0
+  while (i<abstractHand.length){
+    var currentCard = abstractHand[i];
+    if (currentCard.rank >=2 && currentCard.rank <=10){
+      sum = sum + currentCard.rank;
+    } else if (currentCard.rank >=11 && currentCard.rank <=13){
+      sum = sum + 10;
+    } else if (currentCard.rank == 1) {
+      numAcesInHand = numAcesInHand + 1
+      sum = sum + 11
+    };
+     i +=1;
+     console.log(abstractHand)
+  };
+  if (sum > blackjackLimit && numAcesInHand > 0){
+  i = 0
+  while (i < numAcesInHand){
+    sum = sum - 10
+    i +=1
+  if (sum <= blackjackLimit) {
+  break;
+};
+};
+};
+console.log(sum);
+return sum;
+};
+// The user decides whether to hit or stand, using the submit button to submit their choice.
+// The user's cards are analysed for winning or losing conditions.
+// The computer decides to hit or stand automatically based on game rules.
+// The game either ends or continues.
+var MODE_DEAL_CARDS = "deal and display player cards and 1 computer card";
+var MODE_HIT_OR_STAND = "get another card or stay"
+var MODE_COMPARE_CARD_SUM = "compare sum";
+var mode = MODE_DEAL_CARDS;
+
+var printCardsInHand = function(abstractHand){
+  var cardsInHandString = '';
+  for(var i=0;i<abstractHand.length;i+=1){
+  cardsInHandString += `[${abstractHand[i].name} of ${abstractHand[i].suit}] `;
+  }
+return cardsInHandString
+};
+
+var playerHandSum = 0
+var computerHandSum = 0
+var computerStandLimit = 17
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  // Deal cards
+if (mode == MODE_DEAL_CARDS) {
+// Show player cards and first computer card
+// Ask player whether they want to hit or stand
+  dealCard(deck);
+  playerHandSum = getHandSum(playerHand);
+  computerHandSum = getHandSum(computerHand);
+  var myOutputValue = `Your cards are ${playerHand[0].name}, ${playerHand[1].name} with a total of ${playerHandSum} and first computer card is ${computerHand[0].name}. Do you want to Hit or Stand?`;
+  mode = MODE_HIT_OR_STAND;
+// If Hit, deal another card to player and check player hand sum
+} else if (mode == MODE_HIT_OR_STAND){
+  if (input == 'hit'){
+    playerHand.push(deck.pop());
+    playerHandSum = getHandSum(playerHand);
+        console.log(playerHandSum)
+        console.log(blackjackLimit);
+        console.log(playerHand);
+// If player hand sum is less than 21, ask if they want to Hit or Stand
+    if (playerHandSum <= blackjackLimit){
+    myOutputValue = `Your cards are ${printCardsInHand(
+      playerHand)} with a total of ${playerHandSum}. Hit or Stand?`;
+      mode = MODE_HIT_OR_STAND;
+// If > than 21, Game over
+    } else if (playerHandSum > blackjackLimit){
+      myOutputValue = `Your card total is ${playerHandSum}. Game over!`;
+    }
+// If player chooses to Stand, check computer Hand.
+// If less than 17, draw another card.
+  } else if (input == 'stand'){
+    for(var ii = 0; ii<= computerHand.length; ii+=1){
```

Comment:
Can consider using another arbitrary variable as an alternative to `ii`. Common practice is to use `j`, then 'k', etc.